### PR TITLE
minor fixes for emulation on host

### PR DESCRIPTION
**Pull request by d-a-v**
This small [fake async library](https://github.com/d-a-v/emuAsync) permits to compile and run ESPUI on a computer (linux, wsl2).
It may ease debugging of the JS part and WS interactions.
This pull-request allows to make it happen.

*edit*: The `tryID` additional test has been introduced because of an infinite loop.

### CODE REVIEW

**Comment by s00500** [comment]
Hey @d-a-v this looks really cool, need to try it out!

Could you add maybe a bit of documentation to the readme about this ? That would be even more awesome!

**Review by s00500 — APPROVED** [review]

**Comment by s00500** [comment]
I will merge this once I did a quick test of it :-)

**Comment by s00500** [comment]
Hm... could not make this work yet, got some linker issues, do you have any tips what I might be missing ?

make[1]: Entering directory '/home/parallels/espuitest/Arduino/tests/host'
Makefile:18: using g++-9
Makefile:54: compiling in native mode
LD  /home/parallels/espuitest/Arduino/tests/host/bin//home/parallels/espuitest/libs/ESPUI/examples/gui/gui
/usr/bin/ld: /home/parallels/espuitest/Arduino/tests/host/bin/fullcore.a(Hash.cpp.o): in function `sha1(unsigned char const*, unsigned int, unsigned char*)':
/home/parallels/espuitest/Arduino/libraries/Hash/src/Hash.cpp:52: undefined reference to `br_sha1_init'
/usr/bin/ld: /home/parallels/espuitest/Arduino/libraries/Hash/src/Hash.cpp:53: undefined reference to `br_sha1_update'
/usr/bin/ld: /home/parallels/espuitest/Arduino/libraries/Hash/src/Hash.cpp:54: undefined reference to `br_sha1_out'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:405: /home/parallels/espuitest/Arduino/tests/host/bin//home/parallels/espuitest/libs/ESPUI/examples/gui/gui] Error 1
make[1]: Leaving directory '/home/parallels/espuitest/Arduino/tests/host'
make: *** [Makefile:400: /home/parallels/espuitest/libs/ESPUI/examples/gui/gui] Error 2

**Comment by d-a-v** [comment]
Sorry about that.
I just updated the test scripts in the fake async lib repository to fix this issue, which was lacking SSL (BearSSL) library building command.
By hand it's 
```shell
(cd ${ESP8266ARDUINO}/tests/host; make FORCE32=0 ssl)
```

File: README.md [hunk 1]
```diff
@@ -564,6 +564,17 @@ If you change something in HTML/CSS/JS and want to create a pull request, please
 do include the minified versions and corresponding C header files in your
 commits. (Do **NOT** commit all the minified versions for the non changed files)
 
+# Experimental note for Development:
```
Comment (iangray001) [inline README.md:567]: Could this be a more descriptive title? 

"Debugging using Host Emulation ", or similar?

Comment (d-a-v) [reply README.md]: Title is changed to 
> Experimental debugging environment using emulation on host

**Comment by s00500** [comment]
Hi @d-a-v, got it running by building the ssl target as you explained today, this is really super cool, thanks for the contribution!

**Pull request merged**